lib/inline_styles_mailer.rb

```diff
@@ -41,6 +40,14 @@ def css_content
       @css_content
     end
 
+    def stylesheets
+      return @stylesheets if defined? @stylesheets
+
+      path = self.name.underscore
+      modified_path = path.gsub(/\//, '/_')
+      @stylesheets = [modified_path == path ? "_#{path}*" :  "#{modified_path}*"]
```

Comment:
I see what you're doing here and I like it. This will cope with paths two levels deep.

```ruby
# Two levels
Foo::Bar.name.underscore.gsub /\//, '/_'
=> 'foo/_bar' # good
```

But we might as well make it work for any number of levels, which the above code doesn't:

```ruby
# Three levels
Foo::Bar::Po.name.underscore.gsub /\//, '/_'
=> 'foo/_bar/_po' # bad
```

This regex will do it and also remove the need for the test because it works with just one level too:

```ruby
# One level
Foo.name.underscore.sub /^(.*\/)*(.*)$/, '\1_\2*'
=> '_foo*' # good
```

```ruby
# Two levels
Foo::Bar.name.underscore.sub /^(.*\/)*(.*)$/, '\1_\2*'
=> 'foo/_bar*' # good
```

```ruby
# Three levels
Foo::Bar::Po.name.underscore.sub /^(.*\/)*(.*)$/, '\1_\2*'
=> 'foo/bar/_po*' # good
```

The method can then be:

```ruby
def stylesheets
  return @stylesheets if defined? @stylesheets
  [self.name.underscore.sub(/^(.*\/)*(.*)$/, '\1_\2*')]
end
```

I'm not sure it's even worth memoizing that method.